common/djangoapps/util/tests/test_course_key_utils.py

```diff
@@ -1,32 +1,46 @@
 """
 Tests for util.course_key_utils
 """
-from nose.tools import assert_equals, assert_raises  # pylint: disable=no-name-in-module
-from util.course_key_utils import from_string_or_404
+from util.course_key_utils import course_key_from_string_or_404
 from opaque_keys.edx.keys import CourseKey
 from django.http import Http404
+import ddt
+import unittest
 
 
-def test_from_string_or_404():
-
-    #testing with split style course keys
-    assert_raises(
-        Http404,
-        from_string_or_404,
-        "/some.invalid.key/course-v1:TTT+CS01+2015_T0"
-    )
-    assert_equals(
-        CourseKey.from_string("course-v1:TTT+CS01+2015_T0"),
-        from_string_or_404("course-v1:TTT+CS01+2015_T0")
+@ddt.ddt
+class TestFromStringOr404(unittest.TestCase):
+    """
+    Base Test class for course_key_from_string_or_404 utility tests
+    """
+    @ddt.data(
+        ("/some.invalid.key/course-v1:TTT+CS01+2015_T0", "course-v1:TTT+CS01+2015_T0"),  # split style course keys
+        ("/some.invalid.key/TTT/CS01/2015_T0", "TTT/CS01/2015_T0"),  # mongo style course keys
     )
+    def test_from_string_or_404(self, (invalid_course_key, valid_course_key)):
+        """
+        Tests course_key_from_string_or_404 for valid and invalid split style course keys and mongo style course keys.
+        """
+        self.assertRaises(
+            Http404,
+            course_key_from_string_or_404,
+            invalid_course_key,
+        )
+        self.assertEquals(
+            CourseKey.from_string(valid_course_key),
+            course_key_from_string_or_404(valid_course_key)
+        )
 
-    #testing with mongo style course keys
-    assert_raises(
-        Http404,
-        from_string_or_404,
-        "/some.invalid.key/TTT/CS01/2015_T0"
-    )
-    assert_equals(
-        CourseKey.from_string("TTT/CS01/2015_T0"),
-        from_string_or_404("TTT/CS01/2015_T0")
+    @ddt.data(
+        "/some.invalid.key/course-v1:TTT+CS01+2015_T0",  # split style invalid course key
+        "/some.invalid.key/TTT/CS01/2015_T0"  # mongo style invalid course key
     )
+    def test_from_string_or_404_with_message(self, course_string):
+        """
+        Tests course_key_from_string_or_404 with exception message for split style and monog style invalid course keys.
+        :return:
+        """
+        try:
+            course_key_from_string_or_404(course_string, message="Invalid Keys")
```

Comment:
if `course_key_from_string` doesn't return a 404, then this test will pass. Use a pattern [like this](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises) instead:
```python
with self.assertRaises(Http404) as exception:
    course_key_from_string_or_404(course_string, message="Invalid Keys")

self.assertEquals(str(exception), "Invalid Keys")
```